riscv/processor.cc

```diff
@@ -605,18 +605,22 @@ void processor_t::set_virt(bool virt)
        * we should sync Guest/VM FS, VS, and XS state with Host FS,
        * VS, and XS state.
        */
-       if ((state.mstatus & SSTATUS_FS) == SSTATUS_FS) {
-         state.vsstatus |= SSTATUS_FS;
-         state.vsstatus |= (xlen == 64 ? SSTATUS64_SD : SSTATUS32_SD);
-       }
-       if ((state.mstatus & SSTATUS_VS) == SSTATUS_VS) {
-         state.vsstatus |= SSTATUS_VS;
-         state.vsstatus |= (xlen == 64 ? SSTATUS64_SD : SSTATUS32_SD);
-       }
-       if ((state.mstatus & SSTATUS_XS) == SSTATUS_XS) {
-         state.vsstatus |= SSTATUS_XS;
+      if (supports_extension('F')) {
+        state.vsstatus &= ~SSTATUS_FS;
+        state.vsstatus |= (state.mstatus & SSTATUS_FS);
+      }
```

Comment:
The FS field exists even without the F extension (see https://github.com/riscv/riscv-isa-manual/blob/bb9e0561ce5df5e2ed65ebc2f4b253b59f534e82/src/machine.tex#L916) so you should remove the `supports_extension('F')` check.

(By contrast, it's correct to have the `supports_extension('V')` check for VS.)

Reply:
Sure, I will update as-per your suggestion.